src/caret_analyze/architecture/architecture.py

```diff
@@ -139,6 +140,16 @@ def path_names(self) -> Tuple[str, ...]:
     def communications(self) -> Tuple[CommunicationStructValue, ...]:
         return tuple([v.to_value() for v in self._communications])
 
+    @property
+    def publishers(self) -> Tuple[PublisherStructValue, ...]:
```

Comment:
I have a question.
`Architecture.publishers/subscriptions` returns as Tuple and `Application.publishers/subscriptions` returns as List.
Why is there difference?

Reply:
Architecture class returns **immutable** instances (tuple).
Application class returns **mutables** instances (list).

https://tier4.github.io/CARET_doc/latest/design/software_architecture/caret_analyze/#architecture
> All information retrieved from the Architecture file is of type ValueObject, which is suitable for interfacing information between other packages.